fhir-persistence-blob-app/README.md

```diff
@@ -0,0 +1,225 @@
+
+## Configurating the IBM FHIR Server to use Payload Offloading with Azure Blob
+
+To enable payload offloading using Azure Blob storage, complete the steps summarized below:
+
+1. Pick a container name to use for each tenant/datasource combination;
+2. Create a database.properties file with connection information for the FHIRSERVER user (not FHIRADMIN);
+3. Configure payload offloading in the main `default/fhir-server-config.json` file;
+4. Configure payload offloading in each tenant-specific `fhir-server-config.json` file;
+5. Create the container using the Azure Blob user interface, the Azure Blob API or the IBM FHIR Server fhir-persistence-blob-app-*-cli.jar tool;
+6. (Optional) Periodically run the reconciliation tool to remove any orphan resource records from the offload datastore.
+
+Take the following restrictions into account:
+
+1. Payload offloading is enabled at the server level. Offloading cannot be enabled/disabled on a per-tenant basis. If you want to support offloading for just one tenant, use a different IBM FHIR Server instance;
+2. Payload offloading must be configured prior to ingesting any resource data;
+3. Payload offloading must not be disabled after resource data has been ingested;
+
+### 1. Pick a Container Name
+
+Each tenant/datasource combination requires its own container. Use a name that can be easily identified as belonging to the tenant and datasource, subject to the Azure Blob service naming restrictions.
+
+### 2. Create the `database.properties` file
+
+Create a properties file containing the RDBMS connection information. Note that the user should not be the FHIRADMIN user which is only used by the RDBMS schema creation tool. Use the database user configured in the IBM FHIR Server `datasource.xml` file. Following the principle of least privilege access, this user typically has just the right set of privileges for the application to use objects in the FHIR data (`fhirdata`) schema:
+
+```
+db.host=localhost
+db.port=5432
+db.database=fhirdb
+user=fhirserver
+password=change-password
+currentSchema=fhirdata
+```
+
+### 3. Add Payload Offload Configuration to `default/fhir-server-config.json`
+
+In the main `default/fhir-server-config.json`, configure the `fhirServer/persistence/factoryClassname` as shown below and add a `fhirServer/persistence/payload` block containing the connection information for each datasource you have defined under `fhirServer/persistence/datasources` (typically there is just one datasource called `default`). 
+```
+{
+    "__comment": "FHIR Server configuration",
+    "fhirServer": {
+        ...
+        "persistence": {
+            "factoryClassname": "com.ibm.fhir.persistence.blob.FHIRPersistenceJDBCBlobFactory",
+            "datasources": {
+                ...
```

Comment:
might be worth including the default dsid here just to drive it home:
```suggestion
                "default": {
                ...
```

Reply:
applies for section 4 as well

Reply:
done